### PR TITLE
feat: use default SMTP credential

### PR DESCRIPTION
**Pull request by adityathebe**
Resolves: https://github.com/flanksource/incident-commander/issues/514

### CODE REVIEW

File: notification/shoutrrr.go [hunk 1]
```diff
@@ -10,6 +12,33 @@ import (
...
+	)
+	shoutrrrURL = strings.ReplaceAll(shoutrrrURL, defaultSMTPPrefix, prefix)
+
+	parsedURL, err := url.Parse(shoutrrrURL)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse shoutrrr URL: %w", err)
+	}
+
+	query := parsedURL.Query()
+	query.Set("FromAddress", os.Getenv("SMTP_USER"))
```
Comment (moshloop) [inline notification/shoutrrr.go:35]: I think we are also missing an SMTP_FROM - note the user and from is not necessarily the same (if from not specified then use SMTP_USER)

File: notification/shoutrrr.go [hunk 2]
```diff
@@ -10,6 +12,33 @@ import (
 	"github.com/flanksource/incident-commander/api"
 )
 
+// defaultSMTPPrefix indicates that the shoutrrr URL for smtp should use
+// the system's SMTP credentials.
+const defaultSMTPPrefix = "smtp://auto:auto@auto/"
```
Comment (moshloop) [inline notification/shoutrrr.go:17]:
```suggestion
const defaultSMTPPrefix = "smtp://system/"
```

File: notification/shoutrrr.go (hunk 1 above)
Comment (adityathebe) [reply notification/shoutrrr.go:35]: We seem to be taking SMTP from as a flag. Will use that.

File: notification/shoutrrr.go [hunk 3]
```diff
@@ -2,14 +2,45 @@ package notification
...
+	)
+	shoutrrrURL = strings.ReplaceAll(shoutrrrURL, defaultSMTPPrefix, prefix)
+
+	parsedURL, err := url.Parse(shoutrrrURL)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse shoutrrr URL: %w", err)
+	}
+
+	query := parsedURL.Query()
+	query.Set("FromAddress", utils.Coalesce(mail.FromAddress, os.Getenv("SMTP_FROM"), os.Getenv("SMTP_USER")))
```
Comment (moshloop) [inline notification/shoutrrr.go:37]: Will this not always return mail.FromAddress as it has a default?

File: notification/shoutrrr.go [hunk 4]
```diff
@@ -2,14 +2,45 @@ package notification
...
 	"github.com/containrrr/shoutrrr/pkg/types"
 	"github.com/flanksource/commons/collections"
+	"github.com/flanksource/commons/utils"
 	"github.com/flanksource/incident-commander/api"
+	"github.com/flanksource/incident-commander/mail"
 )
 
+// defaultSMTPPrefix indicates that the shoutrrr URL for smtp should use
+// the system's SMTP credentials.
+const defaultSMTPPrefix = "smtp://system/"
```
Comment (moshloop) [inline notification/shoutrrr.go:19]: Can we create a SendEmail(ctx, title, message, properties) that uses this default? and then replace all other usages of SMTP_* with this method ?

File: notification/shoutrrr.go (hunk 3 above)
Comment (adityathebe) [reply notification/shoutrrr.go:37]: on the rare cases where the flag is explicitly removed, this will handle it.

**Pull request merged**